apps/smart-contracts/core/test/integration/DepositTradeHelper.test.ts

```diff
@@ -549,4 +569,306 @@ describe('=> DepositTradeHelper', () => {
       core.collateral.permit.reset()
     })
   })
+
+  describe('trade testing', () => {
+    let univ3Factory: UniswapV3Factory
+    let positionManager: NonfungiblePositionManager
+    let governanceWstETHSwapParams: IVault.SingleSwapStruct
+    let governanceWstETHFundingParams: IVault.FundManagementStruct
+    let permitForFunder: IDepositTradeHelper.PermitStruct
+    let permitForRecipient: IDepositTradeHelper.PermitStruct
+    let balancerParamsForDeposit: IDepositTradeHelper.OffChainBalancerParamsStruct
+    let uniswapParamsForTrade: IDepositTradeHelper.OffChainTradeParamsStruct
+    const GOVERNANCE_COLLATERAL_SUPPLY = parseEther('1000')
+    const GOVERNANCE_LSTOKEN_SUPPLY = parseEther('1000')
+    snapshotter.setupSnapshotContext('DepositTradeHelper-TradeTesting')
+    before(async () => {
+      // Mint governance wstETH to supply market liquidity for pools
+      const wstethNeededForGovernance = await getBaseTokenAmountForDeposit(
+        core.collateral,
+        GOVERNANCE_COLLATERAL_SUPPLY.add(GOVERNANCE_LSTOKEN_SUPPLY)
+      )
+      const ethNeededForGovernance = await getBalancerSingleSwapQuote(
+        1,
+        wstethNeededForGovernance,
+        weth.address,
+        wsteth.address
+      )
+      governanceWstETHSwapParams = {
+        poolId: WSTETH_ETH_METASTABLE_POOL_ID,
+        kind: 1,
+        assetIn: ZERO_ADDRESS,
+        assetOut: wsteth.address,
+        amount: wstethNeededForGovernance,
+        userData: [],
+      }
+      governanceWstETHFundingParams = {
+        sender: governance.address,
+        fromInternalBalance: false,
+        recipient: governance.address,
+        toInternalBalance: false,
+      }
+      await wstethVault
+        .connect(governance)
+        .swap(
+          governanceWstETHSwapParams,
+          governanceWstETHFundingParams,
+          ethers.constants.MaxUint256,
+          nowPlusMonths(1),
+          {
+            value: ethNeededForGovernance.mul(101).div(100),
+          }
+        )
+      univ3Factory = await attachUniV3Factory(
+        ethers,
+        getPrePOAddressForNetwork('UNIV3_FACTORY', 'arbitrumOne')
+      )
+      positionManager = await attachNonfungiblePositionManager(
+        ethers,
+        getPrePOAddressForNetwork('UNIV3_POSITION_MANAGER', 'arbitrumOne')
+      )
+      await core.mintLSFromBaseToken(
+        governance,
+        governance,
+        GOVERNANCE_LSTOKEN_SUPPLY,
```

Comment:
```suggestion
        GOVERNANCE_COLLATERAL_SUPPLY.div(3),
```
add comment that we divide by 3 because x LONG needs to be matched with x COLLATERAL at 50/50, and repeated again for short, hence there needs to be a 2:1 Collateral:LS ratio